src/NHibernate/Criterion/SimpleProjection.cs

```diff
@@ -26,11 +26,18 @@ public virtual IType[] GetTypes(string alias, ICriteria criteria, ICriteriaQuery
 			return null;
 		}
 
+		// Since v5.3
+		[Obsolete("This method has no more usage in NHibernate and will be removed in a future version.")]
 		public virtual string[] GetColumnAliases(int loc)
 		{
-			return new string[] {"y" + loc + "_"};
+			return new[] {GetColumnAlias(loc, 0)};
 		}
-		
+
+		public virtual string GetColumnAlias(int position, int columnIndex)
```

Comment:
Make this non virtual protected 
Call `GetColumnAliases(int loc)`

Reply:
Done